tools/commands/integration_test.py

```diff
@@ -132,105 +132,115 @@ def _integration_test(plugin_dir, test_targets, timeout):
-            errors.append(
-                'flutter-tizen test integration_test failed, see the output above for details.'
-            )
-            exit_code = 1
-
-        if exit_code == 0:
-            test_results.append(TestResult.success(plugin_name, test_target))
-        else:
-            test_results.append(
-                TestResult.fail(plugin_name, test_target, errors=errors))
+            process.wait()
+
+            if is_timed_out:
+                errors.append(
+                    """Timeout expired. The test may need more time to finish.
+        If you expect the test to finish before timeout, check if the tests 
+        require device screen to be awake or if they require manually 
+        clicking the UI button for permissions.""")
+                test_results.append(
+                    TestResult.fail(plugin_name, test_target, errors=errors))
+                continue
+            if last_line.strip() == 'No tests ran.':
+                # This message occurs when the integration test file exists,
+                # but no actual test code is written in it.
+                test_results.append(
+                    TestResult.fail(plugin_name, test_target, [
+                        'Missing integration tests (use --exclude if this is intentional).'
+                    ]))
+                continue
+            elif last_line.strip().startswith('No devices found'):
+                test_results.append(
+                    TestResult.fail(
+                        plugin_name, test_target,
+                        ['The runner cannot find any devices to run tests.']))
+                continue
+
+            match = re.search(_LOG_PATTERN, last_line.strip())
+            if not match:
+                test_results.append(
+                    TestResult.fail(
+                        plugin_name, test_target,
+                        ['Log message is not formatted correctly.']))
+                continue
+
+            # In some cases, the command returns 0 for failed cases, so we check again
+            # with the last log message.
+            exit_code = process.returncode
+            if match.group(2) == 'All tests passed!':
+                exit_code = 0
+            elif match.group(2) == 'Some tests failed.':
+                errors.append(
+                    'flutter-tizen test integration_test failed, see the output above for details.'
+                )
+                exit_code = 1
+
+            if exit_code == 0:
+                test_results.append(
+                    TestResult.success(plugin_name, test_target))
+            else:
+                test_results.append(
+                    TestResult.fail(plugin_name, test_target, errors=errors))
+    finally:
+        subprocess.run('flutter-tizen clean',
+                       shell=True,
+                       cwd=example_dir,
+                       stderr=subprocess.PIPE,
+                       stdout=subprocess.PIPE)
```

Comment:
You don't need set `stderr` and `stdout` to `subprocess.PIPE` in this case.

Reply:
Right, thanks!